WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -259,7 +265,13 @@ public bool TryGetCoinsByInputPrevOut(OutPoint prevOut, [NotNullWhen(true)] out
 
 	private bool TryGetCoinsByInputPrevOutNoLock(OutPoint prevOut, [NotNullWhen(true)] out HashSet<SmartCoin>? coins)
 	{
-		return CoinsByPrevOuts.TryGetValue(prevOut, out coins);
+		coins = null;
+		if (!TxIdsByPrevOuts.TryGetValue(prevOut, out var txId))
+		{
+			return false;
+		}
+
+		return CoinsByTransactionId.TryGetValue(txId, out coins);
```

Comment:
When we are processing a transaction, we go through all its inputs to check if their `prevOuts` are double spending a transaction that created coins for us. So if we are doing the second lookup, it means that one of the inputs is indeed double spending a transaction. The second lookup is just to know which coins are being double spent.

So you see, we are almost never calling the second `TryGetValue`, but I have an interesting note about it:
-> It's useless to return the coins here. Because of the new factoring of `Undo`, if you check, we are never using the coins, as we will call `Undo` with the txid afterwards, and Undo gives back the coins that were removed anyway. So the second look up is useless, we could simply slightly update the `TransactionProcessor`